packages/cli/src/util/extension/proxy.ts

```diff
@@ -0,0 +1,18 @@
+import httpProxy from 'http-proxy';
+import { createServer, Server } from 'http';
+
+export function createProxy(apiUrl: string, token?: string): Server {
+  const proxy = httpProxy.createProxyServer({
+    target: apiUrl,
+    changeOrigin: true,
+  });
+
+  const server = createServer((req, res) => {
+    if (token && !req.headers.authorization) {
+      req.headers.authorization = `Bearer ${token}`;
+    }
+    proxy.web(req, res);
+  });
+
+  return server;
+}
```

Comment:
I should mention that I didn't test this part sorry, as we're not _yet_ authenticating.